src/main/java/co/cask/hydrator/plugin/spark/dynamic/ScalaSparkCodeExecutor.java

```diff
@@ -0,0 +1,395 @@
+    }
+
+    Type[] typeParams = ((ParameterizedType) rddType).getActualTypeArguments();
+    if (typeParams.length < 1 || !typeParams[0].equals(StructuredRecord.class)) {
+      throw new IllegalArgumentException(errorMessage);
+    }
+  }
+
+  private SQLContext getSQLContext(SparkContext sc) {
+    SQLContext sqlContext = sqlContextThreadLocal.get();
+    if (sqlContext != null && !sqlContext.sparkContext().isStopped()) {
+      return sqlContext;
+    }
+
+    synchronized (this) {
+      sqlContext = sqlContextThreadLocal.get();
+      if (sqlContext == null || sqlContext.sparkContext().isStopped()) {
+        sqlContext = new SQLContext(sc);
+        sqlContextThreadLocal.set(sqlContext);
+      }
+    }
+
+    return sqlContext;
+  }
+
+  private Method getMethod(ClassLoader classLoader, String className) {
+    // Use reflection to load the class and get the transform method
+    try {
+      Class<?> computeClass = classLoader.loadClass(className);
+
+      // Find which method to call
+      Method method = null;
+      for (Class<?>[] paramTypes : ACCEPTABLE_PARAMETER_TYPES) {
+        method = tryFindMethod(computeClass, methodName, paramTypes);
+        if (method != null) {
+          break;
+        }
+      }
+
+      String returnType1 = isVoid ? "Unit" : "RDD[StructuredRecord]";
+      String returnType2 = isVoid ? "Unit" : "DataFrame[StructuredRecord]";
+      if (method == null) {
+        throw new IllegalArgumentException(String.format(
+          "Missing a `%s` method that has signature in one of the following form\n" +
+            "def %s(rdd: RDD[StructuredRecord]) : %s\n" +
+            "def %s(rdd: RDD[StructuredRecord], context: SparkExecutionPluginContext) : %s\n" +
+            "def %s(dataframe: DataFrame) : %s\n" +
+            "def %s(dataframe: DataFrame, context: SparkExecutionPluginContext) : %s",
+          methodName, methodName, returnType1, methodName, returnType1, methodName, returnType2,
+          methodName, returnType2));
+      }
+
+      Type[] parameterTypes = method.getGenericParameterTypes();
+      boolean isDataFrame = isDataFrame(parameterTypes[0]);
+
+      // The first parameter should be of type RDD[StructuredRecord] if it takes RDD
+      if (!isDataFrame) {
+        validateRDDType(parameterTypes[0], String.format(
+          "The first parameter of the '%s' method should have type as 'RDD[StructuredRecord]'", methodName));
+      }
+
+      // If it has second parameter, then must be SparkExecutionPluginContext
+      if (parameterTypes.length == 2 && !SparkExecutionPluginContext.class.equals(parameterTypes[1])) {
+        throw new IllegalArgumentException(String.format(
+          "The second parameter of the '%s' method should have type as SparkExecutionPluginContext", methodName));
+      }
+
+      // The return type of the method must be Unit
+      if (isVoid) {
+        if (!"void".equals(method.getReturnType().getName())) {
```

Comment:
Better compare it with `void.class.equals(method.getReturnType())`

Reply:
ah that's the class. Wasn't expecting all lowercase class.